src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -1188,6 +1188,12 @@ ProcessCLRException(IN     PEXCEPTION_RECORD   pExceptionRecord
             RestoreSOToleranceState();
 #endif
 
+#ifdef TARGET_AMD64
+            // OSes older than Win8 have a bug where RtlUnwindEx passes meaningless ContextFlags to the personality routine in
+            // some cases.
+            pContextRecord->ContextFlags |= CONTEXT_FULL;
```

Comment:
I would prefer setting just the three flags (CONTEXT_CONTROL | CONTEXT_INTEGER | CONTEXT_FLOATING_POINT ) instead of the CONTEXT_FULL. We don't care about the debug registers and segment registers that are also in the CONTEXT_FULL.

Reply:
I believe `CONTEXT_FULL` only contains those three bits, `CONTEXT_ALL` contains some other bits. Let me know if you would prefer to be explicit though, I can change it.

Reply:
I am sorry for mistaken the CONTEXT_FULL for CONTEXT_ALL. I would prefer being explicit here.